### PR TITLE
BUG - tinker graph computer was not working with analytics

### PR DESCRIPTION
**Pull request by sheldonkhall**
- The main part of this PR is the change of dependency to our version of Tinkerpop
- This has a small patch to the tinkergraph computer
- more details here: https://issues.apache.org/jira/browse/TINKERPOP-1519

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,72 @@
+#!groovy
+//properties([pipelineTriggers([cron('H H/6 * * *')])])
+node('slave1-dev-jenkins') {
```
Comment (fppt) [inline Jenkinsfile]: I am not going to even pretend I understand this file. What's this for? Is it related to this PR?

File: grakn-graql/src/main/java/ai/grakn/graql/internal/analytics/ShortestPathVertexProgram.java [hunk 2]
```diff
@@ -249,6 +249,8 @@ public boolean terminate(final Memory memory) {
                 if (predecessors.length > 2) {
                     memory.set(MIDDLE, predecessors[2]);
                 }
+                //todo: be careful in tinkergraph as things set here cannot be got
```
Comment (fppt) [inline grakn-graql/src/main/java/ai/grakn/graql/internal/analytics/ShortestPathVertexProgram.java]: Nitpick `//TODO:`

File: grakn-test/src/test/java/ai/grakn/test/graql/analytics/AnalyticsTest.java [hunk 3]
```diff
@@ -74,9 +74,6 @@ public void setUp() {
 
     @Test
     public void testInferredResourceRelation() throws InvalidGraphException {
-        // TODO: Fix on TinkerGraphComputer
```
Comment (fppt) [inline grakn-test/src/test/java/ai/grakn/test/graql/analytics/AnalyticsTest.java]: These removals must have felt great! :)

File: pom.xml [hunk 4]
```diff
@@ -48,7 +48,7 @@
     <properties>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <java.version>1.8</java.version>
-        <tinkerpop.version>3.1.1-incubating</tinkerpop.version>
```
Comment (fppt) [inline pom.xml]: Cool. But a nitpick for a future PR is to change 

```
<titan.version>1.1.0.grakn-SNAPSHOT</titan.version>
```

to 

```
<titan.version>1.1.0-grakn-SNAPSHOT</titan.version>
```

Or maybe just make that change when we migrate to Janus. Anyway, just noting it here.

File: Jenkinsfile (hunk 1 above)
Comment (sheldonkhall) [inline Jenkinsfile]: sorry this was not supposed to be in the PR I will remove it

File: grakn-graql/src/main/java/ai/grakn/graql/internal/analytics/ShortestPathVertexProgram.java (hunk 2 above)
Comment (sheldonkhall) [inline grakn-graql/src/main/java/ai/grakn/graql/internal/analytics/ShortestPathVertexProgram.java]: will change seeing as I have to push a change anyway

File: pom.xml (hunk 4 above)
Comment (sheldonkhall) [inline pom.xml]: ok, this requires redeploying so will put it off

**Pull request merged**